sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -98,6 +100,14 @@ private object PostgresDialect extends JdbcDialect with SQLConfHelper {
     case _ => None
   }
 
+  override def convertJavaTimestampToTimestampNTZ(t: Timestamp): LocalDateTime = {
+    t.toLocalDateTime
```

Comment:
After the change refer: https://github.com/apache/spark/pull/40678/files#r1162437868
We can update with `DateTimeUtils.localDateTimeToMicros(t.toLocalDateTime)` here.